rskj-core/src/test/java/co/rsk/remasc/RemascFederationProviderTest.java

```diff
@@ -2,43 +2,110 @@
 
 import co.rsk.blockchain.utils.BlockGenerator;
 import co.rsk.config.BridgeRegTestConstants;
+import co.rsk.core.RskAddress;
+import co.rsk.peg.*;
 import co.rsk.test.builders.BlockChainBuilder;
+import co.rsk.util.HexUtils;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
 import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
 import org.ethereum.core.Blockchain;
 import org.ethereum.core.Genesis;
+import org.ethereum.crypto.ECKey;
+import org.ethereum.vm.PrecompiledContracts;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+import static org.mockito.Mockito.*;
+
 /**
  * Created by ajlopez on 14/11/2017.
  */
 class RemascFederationProviderTest {
+
+    private static final String BTC_PUBLIC_KEY = "0x03a5e32151f974c35c5d08af36a8d6af0593248e8e4adca7ed0148192eb2f5d1bf";
+    private static final String RSK_PUBLIC_KEY = "0x02f3b5fb3121932db9450121b0a37f3f051dc8b3fd5f1ea5e7cf726947d8f69c28";
+
     @Test
     void getDefaultFederationSize() {
         RemascFederationProvider provider = getRemascFederationProvider();
         Assertions.assertEquals(3, provider.getFederationSize());
     }
 
     @Test
-    void getFederatorAddress() {
-        RemascFederationProvider provider = getRemascFederationProvider();
+    void getFederatorAddress_preRSKIP415_returnsRskAddressDerivedFromBtcPublicKey() {
+
+        int federatorIndex = 0;
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP415)).thenReturn(false);
+
+        FederationSupport federationSupport =  mock(FederationSupport.class);
+
+        byte[] btcPublicKey = HexUtils.strHexOrStrNumberToByteArray(BTC_PUBLIC_KEY);
 
-        byte[] address = provider.getFederatorAddress(0).getBytes();
+        when(federationSupport.getFederatorBtcPublicKey(federatorIndex))
+                .thenReturn(btcPublicKey);
+
+        RemascFederationProvider provider = new RemascFederationProvider(activations, federationSupport);
+
+        RskAddress actualRskAddress = provider.getFederatorAddress(federatorIndex);
+        RskAddress expectedRskAddress = new RskAddress(ECKey.fromPublicOnly(btcPublicKey).getAddress());
+
+        Assertions.assertEquals(expectedRskAddress, actualRskAddress);
+        verify(federationSupport, never()).getFederatorPublicKeyOfType(federatorIndex, FederationMember.KeyType.RSK);
+        verify(federationSupport, times(1)).getFederatorBtcPublicKey(federatorIndex);
+
+    }
+
+    @Test

```

Comment:
Think we could have these 2 as single parameterized test?

Reply:
but its doing different checks right? 

```
verify(federationSupport, never()).getFederatorPublicKeyOfType(federatorIndex, FederationMember.KeyType.RSK);
verify(federationSupport, times(1)).getFederatorBtcPublicKey(federatorIndex);
```

versus

```
verify(federationSupport, times(1)).getFederatorPublicKeyOfType(federatorIndex, FederationMember.KeyType.RSK);
verify(federationSupport, never()).getFederatorBtcPublicKey(federatorIndex);
```

Reply:
That being said, i would invert the order in the second test:

```
verify(federationSupport, never()).getFederatorBtcPublicKey(federatorIndex);
verify(federationSupport, times(1)).getFederatorPublicKeyOfType(federatorIndex, FederationMember.KeyType.RSK);
```

to see the difference between both tests more clear